crates/sparrow-runtime/src/execute.rs

```diff
@@ -236,3 +240,128 @@ pub async fn execute(
         request.compute_snapshot_config,
     ))
 }
+
+pub async fn materialize(
```

Comment:
Some of the setup code is similar to the `execute` method above, and could be refactored.